addons/sale/report/sale_report_views.xml

```diff
@@ -77,7 +78,9 @@
                     <filter string="Customer" name="Customer" context="{'group_by':'partner_id'}"/>
                     <filter string="Customer Country" name="country_id" context="{'group_by':'country_id'}"/>
                     <filter string="Customer Industry" name="industry_id" context="{'group_by':'industry_id'}"/>
-                    <filter string="Product" name="Product" context="{'group_by':'product_id'}"/>
+                    <filter string="Product" name="Product" context="{'group_by':'product_tmpl_id'}"/>
+                    <filter string="Product Variant" name="Product Variant" context="{'group_by':'product_id'}"
```

Comment:
```suggestion
                    <filter string="Product" name="product_tmpl_id" context="{'group_by': 'product_tmpl_id'}"/>
                    <filter string="Product Variant" name="product_id" context="{'group_by': 'product_id'}"
```
The name is a technical attribute for hooking purposes, and ideally refers to the field instead of a description ;)